### PR TITLE
CB-10696 iOS: Encode target path with spaces

### PR DESCRIPTION
**Pull request by matrosov-nikita**
### Platforms affected
iOS

### What does this PR do?
Encodes target path for downloading process

### What testing has been done on this change?
Auto-test

### Checklist
- [x] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [x] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [x] Added automated test coverage as appropriate for this change.

### CODE REVIEW

**Comment by cordova-qa** [comment]
Cordova CI Build has completed successfully.

**Commit**     - [Link](https://github.com/apache/cordova-plugin-file-transfer/pull/175/commits/7c32fcb6b9937fb6e97f50790493cb532cbd6ad4)
**Dashboard** - [Link](http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62/)

| Builder Name  | Console Output | Test Report | Device Logs  |
|     :---:     |     :---:      |   :---:     |     :---:    |
| [Windows 10  Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=windows-10-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=windows-10-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=windows-10-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=windows-10-store/artifact/) |
| [iOS 9.3]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=ios-9.3/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=ios-9.3/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=ios-9.3/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=ios-9.3/artifact/) |
| [iOS 10.0]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=ios-10.0/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=ios-10.0/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=ios-10.0/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=ios-10.0/artifact/) |
| [Android 4.4]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=android-4.4/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=android-4.4/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=android-4.4/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=android-4.4/artifact/) |
| [Android 5.1]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=android-5.1/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=android-5.1/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=android-5.1/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/62//PLATFORM=android-5.1/artifact/) |
 265 tests run, 13 skipped, 0 failed.

File: src/ios/CDVFileTransfer.m [hunk 1]
```diff
@@ -440,13 +440,23 @@ - (void)download:(CDVInvokedUrlCommand*)command
         targetURL = [[self.commandDelegate getCommandInstance:@"File"] fileSystemURLforLocalPath:target].url;
     } else {
         targetURL = [NSURL URLWithString:target];
+
+        if (targetURL == nil) {
+            NSString* targetUrlTextEscaped = [target stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
```
Comment (blankg) [inline src/ios/CDVFileTransfer.m]: stringByAddingPercentEscapesUsingEncoding is deprecated, would be better to use [target stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLFragmentAllowedCharacterSet]]

Comment (matrosov-nikita) [inline src/ios/CDVFileTransfer.m]: @blankg, thanks, updated

**Comment by cordova-qa** [comment]
Cordova CI Build has one or more failures. 

Please note that a lot of tests are currently disabled on CI. To make sure everything works, please run tests locally via cordova-paramedic:
https://github.com/apache/cordova-paramedic

**Commit**     - [Link](https://github.com/apache/cordova-plugin-file-transfer/pull/175/commits/5f8e159ddbb8230b8d31606fa6a608c2e009a676)
**Dashboard** - [Link](http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69/)

| Builder Name  | Console Output | Test Report | Device Logs  |
|     :---:     |     :---:      |   :---:     |     :---:    |
| [Windows 10  Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=windows-10-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=windows-10-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=windows-10-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=windows-10-store/artifact/) |
| [iOS 9.3]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=ios-9.3/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=ios-9.3/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=ios-9.3/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=ios-9.3/artifact/) |
| [iOS 10.0]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=ios-10.0/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=ios-10.0/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=ios-10.0/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=ios-10.0/artifact/) |
| [Android 4.4]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=android-4.4/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=android-4.4/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=android-4.4/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=android-4.4/artifact/) |
| [Android 5.1]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=android-5.1/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=android-5.1/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=android-5.1/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-file-transfer-pr/69//PLATFORM=android-5.1/artifact/) |
 265 tests run, 13 skipped, 12 failed.

**Comment by alsorokin** [comment]
Let there be tests

**Pull request merged**